src/Tribe/Linked_Posts.php

```diff
@@ -760,54 +813,88 @@ public function handle_submission_by_post_type( $event_id, $linked_post_type, $s
 			$post_status = $event_post_status;
 
 			/**
-			 * Filters the ID (default null) for creating posts from the event edit page
+			 * Filters the ID (default null) for creating posts from the event edit page.
 			 *
-			 * @param string $id Post type id index
-			 * @param array $data Data for submission
-			 * @param string $linked_post_type Post type
-			 * @param string $post_status Post status
-			 * @param int $event_id Post ID of the post the post type is attached to
+			 * @param string $id               Post type ID index.
+			 * @param array  $data             Data for submission.
+			 * @param string $linked_post_type Post type.
+			 * @param string $post_status      Post status.
+			 * @param int    $event_id         Post ID of the Event the Linked Post is attached to.
 			 */
 			$id = apply_filters( 'tribe_events_linked_post_create_' . $linked_post_type, null, $data, $linked_post_type, $post_status, $event_id );
 
 			/**
-			 * Filters the ID (default null) for creating posts from the event edit page
+			 * Filters the ID (default null) for creating posts from the event edit page.
 			 *
-			 * @param string $id Post type id index
-			 * @param array $data Data for submission
-			 * @param string $linked_post_type Post type
-			 * @param string $post_status Post status
-			 * @param int $event_id Post ID of the post the post type is attached to
+			 * @param string $id               Post type id index.
+			 * @param array  $data             Data for submission.
+			 * @param string $linked_post_type Post type.
+			 * @param string $post_status      Post status.
+			 * @param int    $event_id         Post ID of the Event the Linked Post is attached to.
 			 */
 			$id = apply_filters( 'tribe_events_linked_post_create', $id, $data, $linked_post_type, $post_status, $event_id );
 
-			if ( $id ) {
-				$linked_posts[] = $id;
+			if ( ! empty( $id ) ) {
+				$post_ids_to_link[] = $id;
 			}
 		}
 
-		// if we don't allow multiples, make sure there's only 1
-		if ( ! $this->allow_multiple( $linked_post_type ) && count( $linked_posts ) > 1 ) {
-			$linked_posts = array( $linked_posts[0] );
-		}
+		$post_ids_to_link = array_map( 'absint', $post_ids_to_link );
+		$post_ids_to_link = array_filter( $post_ids_to_link );
+		$post_ids_to_link = array_unique( $post_ids_to_link );
 
-		// if we allow multiples and there is more then one save current order
-		if ( $this->allow_multiple( $linked_post_type ) ) {
-			$this->order_linked_posts( $event_id, $linked_post_type, $submission[ $linked_post_type_id_field ] );
+		// If we do not allow multiples for this post type, ignore all but the first.
+		if (
+			! $this->allow_multiple( $linked_post_type )
+			&& 1 < count( $post_ids_to_link )
```

Comment:
I'm not sure about this conditional but you are testing if `$post_ids_to_link` is empty?

Reply:
example: if multiple venue post ID's but the venue post type doesn't allow multiples... then just get the first venue ID in the array of multiple

This check isn't new, but it is a new way to do it.

Reply:
Okay I see. I think using `empty` would improve this one a bit for a couple of reasons.

1. Using `empty( $post_ids_to_link )` provides a more easier way to spot what's going instead of `1 < count( )` 
2. Decreases complexity as removes 2 operations: count and compare into just compare. 

Let me know what do you think?